include/eve/detail/shuffle_v2/simd/x86/shuffle_l4_l5.hpp

```diff
@@ -48,13 +48,34 @@ shuffle_l4_l5_x86_put_u64x2_in_position(P, fixed<G>, wide<T, N> x)
   {
     constexpr auto p0 = get<0>(*P::shuffle_16in16);
     constexpr auto p1 = get<1>(*P::shuffle_16in16);
-    auto [r0, l0] = shuffle_v2_core(x, eve::lane<G>, idxm::to_pattern<p0>());
-    auto [r1, l1] = shuffle_v2_core(r0, eve::lane<G>, idxm::to_pattern<p1>());
+    auto [r0, l0]     = shuffle_v2_core(x, eve::lane<G>, idxm::to_pattern<p0>());
+    auto [r1, l1]     = shuffle_v2_core(r0, eve::lane<G>, idxm::to_pattern<p1>());
 
     return kumi::tuple {r1, idxm::add_shuffle_levels(l0, l1)};
   }
 }
 
+template<typename P, typename T, typename N, std::ptrdiff_t G>
+EVE_FORCEINLINE auto
+shuffle_l4_l5_x86_slide_less_than_16(P, fixed<G>, wide<T, N> x)
+{
+  constexpr auto no = kumi::tuple {no_matching_shuffle, eve::index<-1>};
+  // Coudn't figure out how to generalize well
+  // only slide left for now
+  // No masking 0s on avx512
+  if constexpr( current_api < avx2 ) return no;
+  else if constexpr ( constexpr auto slide = idxm::is_slide_left(P::idxs) )
+  {
+    static_assert(G == 1, "verifying assumptions");
```

Comment:
if the static_assert is that empty of infos, why not just use the single parameter static_assert and not put a reason ?

Reply:
Well, if it fails for a user, I want them to see that it's internal failure.

Also, now looking at it, this message is bad for that

Reply:
I'm OK if we go with a better message. I usually try to include "[EVE]" in the message so we know where it comes from